File: TripsServer/Trips/Entities/Place.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Trips.Entities
{
    public class Place
    {
        public int Id { get; set; }
        public PlaceKind? Kind { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        [Obsolete("No more. Write location to Latitude/Longitude fields. This field will be removed.")]
        public string Location { get; set; }
        public double? Latitude { get; set; }
        public double? Longitude { get; set; }
        public Region Region { get; set; }
        public User AddedBy { get; set; }
        public DateTime? AddedDate { get; set; }
        public User ChangedBy { get; set; }
        public DateTime? ChangedDate { get; set; }

        public DateTime? DiscoveryDate { get; set; }
        public PlaceAccessibility? Accessibility { get; set; }
        public PlaceAccessibility? NearestAccessibility { get; set; }
        public PlacePopularity? Popularity { get; set; }
        public PlaceCapacity? Capacity { get; set; }
        public bool IsXBApproved { get; set; }

        public Picture TitlePicture { get; set; }
        public Gallery Gallery { get; set; }

        public ICollection<Visit> Visits { get; set; }
    }
}

File: trips-client/src/app/models/place.ts
import { UserHeader } from './user-header';
import { Gallery } from './gallery';
import { Picture } from './picture';

export enum PlaceKind {
    LAKE = 1,
    DAM = 2,
    RIVER = 3,
    TOWN = 4,
    RUINS = 5,
    TOURISTATTRACTION = 6,
    ABANDONED = 7
}

export enum PlaceAccessibility {
    ASPHALTGRAVEL = 1,
    DIRTROADFLAT = 2,
    DEEPTRACKSPUDDLES = 3,
    TRACTORONLY = 4,
    NOROAD = 5
}

export enum PlacePopularity {
    ALWAYSFREE = 1,
    SOMETIMESOCCUPIED = 2,
    MOSTPROBABLYOCCUPIED = 3,
    CROWDED = 4
}

export enum PlaceCapacity {
    ZERO = 0,
    SINGLE = 1,
    SEVERAL = 2,
    NUMEROUS = 3
}

export class Place {
    id: number;
    kind: PlaceKind;
    name: string;
    description: string;
    location: string;
    latitude: number;
    longitude: number;
    addedBy: UserHeader;
    addedDate: Date;
    changedBy: UserHeader;
    changedDate: Date;

    discoveryDate: Date;
    accessibility: PlaceAccessibility;
    nearestAccessibility: PlaceAccessibility;
    popularity: PlacePopularity;
    capacity: PlaceCapacity;
    isXBApproved: boolean;

    titlePicture: Picture;
    gallery: Gallery;
}